File: java-patterns/src/main/java/behavioral/strategy/Cache.java
package behavioral.strategy;

import lombok.Setter;

/**
 * Usage: Context <br/>
 * Date: 2023/4/30 21:05 <br/>
 *
 * @author <a href="mailto:hanzhang2566@foxmail.com">hanzhang</a>
 */
public class Cache {
    @Setter
    private Algorithm algorithm;

    public Cache(Algorithm algorithm) {
        this.algorithm = algorithm;
    }

    public void delete() {
        algorithm.delete();
    }
}

File: go-patterns/behavioral/strategy/cache.go
package strategy

// Cache context
type Cache struct {
	algorithm Algorithm
}

// NewCache getContext
func NewCache(a Algorithm) *Cache {
	return &Cache{
		algorithm: a,
	}
}

func (c *Cache) SetAlg(a Algorithm) {
	c.algorithm = a
}

func (c *Cache) Delete() {
	c.algorithm.Delete(c)
}
